Sources/TuistCache/Cache/CacheRemoteStorage.swift

```diff
@@ -82,6 +91,15 @@ final class CacheRemoteStorage: CacheStoring {
 
     // MARK: - Private
 
+    private func zipFlow(downloadedArchive: AbsolutePath, hash: String) throws -> AbsolutePath {
+        let zipPath = try FileHandler.shared.changeExtension(path: downloadedArchive, to: "zip")
+        let archiver = fileArchiver(for: zipPath)
+        let archiveDestination = Environment.shared.xcframeworksCacheDirectory.appending(component: hash)
+        try archiver.unzip(to: archiveDestination)
+        let folderContent = try FileHandler.shared.contentsOfDirectory(archiveDestination)
+        return folderContent.filter { FileHandler.shared.isFolder($0) }.first ?? archiveDestination
```

Comment:
The logic here might not be very obvious. What I understand by reading it is that we are trying to get the path to the .xcframework that lives inside the cache directory. If that's the case, I'd be more accurate and say:
- I expect the directory to contain a directory with a .xcframework extension. If that's the case, I'll return it.
- If the .xcframework doesn't exist, I'd flag the directory as invalid, delete it, and return nil. Returning `arhiveDestination` to the consumer of the method might lead to unexpected behaviors that someone will have troubles to debug in the future.

Reply:
Fair enough. Corrected.